### PR TITLE
Tyreid App for Bangle.js 2

### PR DESCRIPTION
**Pull request by th10111**
Hello, 
This is a Bluetooth war-driving app for the Bangle.js 2. 
It passively logs bluetooth packets with the current time and GPS location. 
Hopefully some people might find it useful/fun! 
Regards, 
Tom

### CODE REVIEW

**Comment by gfwilliams** [comment]
Looks great - thanks! The pre-merge check is failing though because:

* There's no `ChangeLog` - just copy one from the example app and make sure the last version listed is 0.02
* There's no icon file - not sure if you still have the original, but it needs to be a png file (as well as the JS) and that needs referencing from the metadata's 'icon' field (not the .img)

If you wanted to be able to get the file off a bit more easily you can add an 'interface' file to the app loader. The tutorial here basically does exactly what you want - just copy/paste and change the filename: https://www.espruino.com/Bangle.js+Storage#reading-with-the-app-loader

File: apps/Tyreid/app.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    packet.data,
+    packet.serviceData,
+    packet.manufacturerData
+  ];
+  // Write data to the file (including a new line)
+  file.write(new_data.join(",")+"\n");
+  
+  if (num_bt_devices < 99) {
+    if (!bt_id_arr.includes(mac)) { // if device id has not been recorded
+      bt_id_arr[num_bt_devices] = mac; // note the id
```
Comment (bobrippling) [inline apps/Tyreid/app.js:84]:
```suggestion
      bt_id_arr.push(mac); // note the id
```

File: apps/Tyreid/app.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  marker();
+}, BTN1, {repeat:true});
+
+WIDGETS["widTyreid"]={
+		area:"tl", // tl (top left), tr (top right), bl (bottom left), br (bottom right)
+		width: 24, // width of the widget
+		draw: function() {
+      let disp_dev_val = "-";
+
+      if (num_bt_devices < 99) {
```
Comment (bobrippling) [inline apps/Tyreid/app.js:211]:
```suggestion
      if (bt_id_arr.length < 99) {
```

**Review by bobrippling — COMMENTED** [review]
Also, if you find it useful, the `num_bt_devices` variable can be replaced by getting the array length from `bt_id_arr`, I've left a few example changes that would bring this about :)

**Comment by gfwilliams** [comment]
Any chance of fixing what we'd asked, or shall I close this PR?

**Comment by th10111** [comment]
Hi,

Sorry Gordon, I haven't had a lot of time to work on this recently - but hopefully the required changes are there now.

In the future I would like to spend some time working on an interface file, and possibly some code to perform some analysis. 

Also, thank you Rob for your suggestion; unfortunately I haven't had time to change and fully test your suggestion, but if I get some time in the future I will give it a go.

Thanks both,

Tom

**Comment by bobrippling** [comment]
No problem, thanks for those final changes! I tweaked `metadata.json` to bring it into line with your changelog :)

**Pull request merged**